### PR TITLE
Adds internal state documentation

### PR DESCRIPTION
**Pull request by fernandosouza**
I'm not sure if this is the right place to write this documentation once State class don't know about this attribute, but it seems clearer for users once they will work on the STATE object.

### CODE REVIEW

**Comment by fernandosouza** [comment]
@mairatma, could you take a look, please?

File: src/public/docs/state.soy [hunk 1]
```diff
@@ -105,6 +106,30 @@ console.log(obj.config.foo); // Prints 'foo'{/literal}
 /**
  * Content added after a State example.
  */
+{template .internalState}
+	<h2>Internal States</h2>
+
+	<p>
+		Sometimes you want a state to take its initial value from a parent
+		component but you don't want to give to the parent component the power
+		of updating the state value keeping the change logic internally.
```
Comment (brunobasto) [inline src/public/docs/state.soy]: Sentence looks very long. Maybe a comma after

> ... to give to the parent component the power of updating the state value

?

Comment (fernandosouza) [reply src/public/docs/state.soy]: Updated

**Comment by mairatma** [comment]
Nice,  it's important to document this feature. I think it's still hard to understand what `internal` means though. Might be good to link to some jsfiddle example showing how they work differently maybe. It's already useful as it is though.

**Comment by fernandosouza** [comment]
Thx @mairatma

**Comment by fernandosouza** [comment]
Hey guys, do you have any other observation on this pull request?
cc. @eduardolundgren, @jbalsas

**Comment by jbalsas** [comment]
Hey @fernandosouza, @eduardolundgren, would a change in an `internal` property trigger a re-render? I've always assumed it won't. If so, I think this is a key part that should also be documented. 

Even if it were the opposite, I think it should be clearly stated what's the effect (if any) the `internal` attribute has in a state property

**Comment by fernandosouza** [comment]
Hey, @jbalsas. Yes, changes in `internal` states trigger a re-render. The `internal` attribute only means that it can receive its value from the parent component once, in the initialization. After that, only the component itself is able to change its value.

**Comment by jbalsas** [comment]
Nice, I think that's a good thing to document, then ;)

**Comment by fernandosouza** [comment]
Yeah. I'm going to update the doc. Thx. :)

**Comment by mairatma** [comment]
Just one detail, `internal` attributes never receive their values from a parent component, not even on the first time. They're completely internal, just like react's `state`. Parent components should not even know they exist.

**Comment by fernandosouza** [comment]
Thank you for clearing @mairatma. 😃

**Comment by zenorocha** [comment]
@fernandosouza can you update this pull request?

**Comment by fernandosouza** [comment]
Sure, @zenorocha.

**Comment by fernandosouza** [comment]
Updated, @brunobasto, @jbalsas.

File: src/public/docs/state.soy [hunk 2]
```diff
@@ -105,6 +106,36 @@ console.log(obj.config.foo); // Prints 'foo'{/literal}
 /**
  * Content added after a State example.
  */
+{template .internalState}
+	<h2>Internal States</h2>
+
+	<p>
+		You can define a state to be used internally in your component only, just
+		add a <b>internal</b> attribute in the configuration object. By doing this
+		the state will behave exctly as expected, including re-rendering the
```
Comment (brunobasto) [inline src/public/docs/state.soy]: @fernandosouza typo "exctly" should be "exactly"

File: src/public/docs/state.soy [hunk 3]
```diff
@@ -105,6 +106,36 @@ console.log(obj.config.foo); // Prints 'foo'{/literal}
 /**
  * Content added after a State example.
  */
+{template .internalState}
+	<h2>Internal States</h2>
+
+	<p>
+		You can define a state to be used internally in your component only, just
+		add a <b>internal</b> attribute in the configuration object. By doing this
```
Comment (brunobasto) [inline src/public/docs/state.soy]: add **an** internal state

File: src/public/docs/state.soy [hunk 4]
```diff
@@ -105,6 +106,36 @@ console.log(obj.config.foo); // Prints 'foo'{/literal}
...
+		the state will behave exctly as expected, including re-rendering the
+		component. The difference is, internal states can not be accessed by parent
+		components.
+	</p>
+	<p>
+		This is not necessary for JSX components, since Metal JSX components
+		has its own State Manager implementation. For that reason, two static properties
```
Comment (brunobasto) [inline src/public/docs/state.soy]: have instead of has

**Comment by fernandosouza** [comment]
@brunobasto, updated

**Comment by jbalsas** [comment]
Hey @fernandosouza, same as with #53, could you directly send this to https://github.com/metal/metaljs.com/pull/54 or work with @blzaugg so this makes it into the new site?

**Comment by fernandosouza** [comment]
Sure, @jbalsas. 😃

**Comment by blzaugg** [comment]
Added with: https://github.com/metal/metaljs.com/pull/56/commits/ae796b565a496fe278cc4275843cdb1d96bc5172

**Pull request closed without merging**